src/main/java/seedu/intern/logic/commands/EditAllCommand.java

```diff
@@ -0,0 +1,171 @@
+package seedu.intern.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.intern.logic.parser.CliSyntax.PREFIX_STATUS;
+import static seedu.intern.model.Model.PREDICATE_SHOW_ALL_PERSONS;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import seedu.intern.commons.util.CollectionUtil;
+import seedu.intern.logic.commands.exceptions.CommandException;
+import seedu.intern.model.Model;
+import seedu.intern.model.applicant.Applicant;
+import seedu.intern.model.applicant.ApplicationStatus;
+import seedu.intern.model.applicant.Course;
+import seedu.intern.model.applicant.Email;
+import seedu.intern.model.applicant.Grade;
+import seedu.intern.model.applicant.GraduationYearMonth;
+import seedu.intern.model.applicant.Institution;
+import seedu.intern.model.applicant.Name;
+import seedu.intern.model.applicant.Phone;
+import seedu.intern.model.skills.Skill;
+
+/**
+ * Edits the details of all currently displayed applicants in Intern Watcher.
+ */
+public class EditAllCommand extends Command {
+
+    public static final String COMMAND_WORD = "editAll";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Updates all currently displayed applicants with the"
+            + "provided fields."
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: "
+            + "[" + PREFIX_STATUS + "STATUS]\n";
+
+    // TODO: update this message
+    public static final String MESSAGE_EDIT_ALL_SUCCESS = "Edited currently displayed applicants: ";
+    public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.\n";
+
+    private final EditAllDescriptor editAllDescriptor;
+
+    /**
+     * @param editAllDescriptor details to update all applicants with
+     */
+    public EditAllCommand(EditAllDescriptor editAllDescriptor) {
+        this.editAllDescriptor = new EditAllDescriptor(editAllDescriptor);
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Applicant> lastShownList = model.getFilteredPersonList();
+        int addSuccesses = 0;
+
+        // TODO: Catch failures to update?
+        for (Applicant applicantToEdit : lastShownList) {
+            Applicant editedApplicant = createEditedApplicant(applicantToEdit, editAllDescriptor);
+            model.setApplicant(applicantToEdit, editedApplicant);
+            addSuccesses++;
+        }
+
+        String result = addSuccesses + " of " + lastShownList.size() + " applicants updated\n";
+
+        model.updateFilteredApplicantList(PREDICATE_SHOW_ALL_PERSONS);
```

Comment:
Do we want to show the entire list here or do we actually want to show the same filtered list as before we update the details?